deep-learning/src/main/python/synapse/ml/dl/sentence_embedding_transformer.py

```diff
@@ -0,0 +1,498 @@
+            ):
+                sentences_batch = sentences_sorted[
+                    start_index : start_index + batch_size
+                ]
+                features = self.tokenize(sentences_batch)
+                features = batch_to_device(features, device)
+
+                # Perform forward pass and gather embeddings
+                with torch.no_grad():
+                    out_features = self(features)
+
+                    if output_value == "token_embeddings":
+                        embeddings = []
+                        for token_emb, attention in zip(
+                            out_features[output_value], out_features["attention_mask"]
+                        ):
+                            last_mask_id = len(attention) - 1
+                            while (
+                                last_mask_id > 0 and attention[last_mask_id].item() == 0
+                            ):
+                                last_mask_id -= 1
+                            embeddings.append(token_emb[0 : last_mask_id + 1])
+                    elif output_value is None:
+                        embeddings = []
+                        for sent_idx in range(len(out_features["sentence_embedding"])):
+                            row = {
+                                name: out_features[name][sent_idx]
+                                for name in out_features
+                            }
+                            embeddings.append(row)
+                    else:
+                        embeddings = out_features[output_value]
+                        embeddings = embeddings.detach()
+                        if normalize_embeddings:
+                            embeddings = torch.nn.functional.normalize(
+                                embeddings, p=2, dim=1
+                            )
+                        if convert_to_numpy:
+                            embeddings = embeddings.cpu()
+
+                    all_embeddings.extend(embeddings)
+
+            # Restore original order of sentences
+            all_embeddings = [
+                all_embeddings[idx] for idx in np.argsort(length_sorted_idx)
+            ]
+            if convert_to_tensor:
+                all_embeddings = torch.stack(all_embeddings)
+            elif convert_to_numpy:
+                all_embeddings = np.asarray([emb.numpy() for emb in all_embeddings])
+
+            if input_was_string:
+                all_embeddings = all_embeddings[0]
+
+            return all_embeddings
+
+    def __init__(
+        self,
+        inputCol=None,
+        outputCol=None,
+        useTRTFlag=None,
+        driverOnly=False,
+        batchSize=16,
+        model=None,
+        modelName="intfloat/e5-large-v2",
+        moduleName="e5-large-v2",
+    ):
+        """
+        Initialize the EmbeddingTransformer with input/output columns and optional TRT flag.
+        """
+        super(EmbeddingTransformer, self).__init__()
+        self._setDefault(
```

Comment:
try it on some other models from : https://sbert.net/docs/sentence_transformer/pretrained_models.html